jina/types/document/__init__.py

```diff
@@ -135,6 +135,12 @@ def __init__(self, document: Optional[DocumentSourceType] = None,
                              f'if you are trying to set the content '
                              f'you may use "Document(content=your_content)"') from ex
 
+        if custom_id is None:
```

Comment:
not sure if we should, but would it make sense here to check if `self.id` has already been set? in case this is already created from an existing `documentProto`?

Reply:
Very good point. :+1:

Reply:
Now, that I understood the whole concept better, I think that should be exactly the way of setting `id`. There will be no `custom_id` after my next commit.